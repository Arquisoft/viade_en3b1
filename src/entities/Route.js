import { v4 as uuid } from "uuid";
import RouteElement from "./RouteElement";

class Route {

    /**
     * Constructor for a new Route. 
     * An id will be automatically generated (uuid v4).
     * TotalDistance will also be automatically generated.
     * 
     * RouteElements have name, latitude, longitude and elevation.
     * 
     * @param {String} name Name of the route.
     * @param {Date} date Date of the route.
     * @param {String} description Description of the route.
     * @param {Array} routeElements List of RouteElement objects, which are the trackpoints.
     * @param {Array} comments List of comments
     * @param {Array<String>} media List of urls of media (on the POD)
     */
    constructor(name, description, routeElements, comments, media, date) {
        this.id = uuid().toString();
        this.name = name;
        this.date = date;
        this.description = description;

        this.media = media;
        this.totalDistance = null;

        if (comments === null) {
            this.comments = [];
        } else {
            this.comments = comments;
        }

        if (routeElements[0] instanceof RouteElement) {
            this.routeElements = routeElements;
        } else {
            this.routeElements = generateRouteElements(routeElements);
        }
    }

    getId() {
        return this.id;
    }

    getName() {
        return this.name;
    }

    getDate() {
        return this.date;
    }

    getTime() {
        return this.time;
    }

    getDescription() {
        return this.description;
    }

    getRouteElements() {
        return this.routeElements;
    }

    getTotalDistance() {
        var total = 0;
        var aux = 0;

        for (let i = 1; i < this.routeElements.length; i++) {
            aux = this.routeElements[i].getLatitude() - this.routeElements[i - 1].getLatitude();
            total += this.addDistance(aux);
            aux = this.routeElements[i].getLongitude() - this.routeElements[i - 1].getLongitude();
            total += this.addDistance(aux);
            aux = this.routeElements[i].getElevation() - this.routeElements[i - 1].getElevation();
            total += this.addDistance(aux);
        }

        return total;
    }

    addDistance(aux) {
        if (aux < 0) {
            return -aux;
        } else {
            return aux;
        }
    }

    getComments() {
        return this.comments;
    }

    getMedia() {
        return this.media;
    }

    addMedia(media) {
        this.media.push(media);
    }

    addRouteElement(routeElement) {
        this.routeElements.push(routeElement);
    }

    getJsonLD() {
        let routePointsJson = [];
        this.routeElements.forEach((p) => routePointsJson.push(p.toJsonLatLng()));
        return JSON.stringify(
            {
                "@context": {
                    "@version": 1.1,
                    "comments": {
                        "@container": "@list",
                        "@id": "viade:comments"
                    },
                    "description": {
                        "@id": "schema:description",
                        "@type": "xs:string"
                    },
                    "media": {
                        "@container": "@list",
                        "@id": "viade:media"
                    },
                    "name": {
                        "@id": "schema:name",
                        "@type": "xs:string"
                    },
                    "points": {
                        "@container": "@list",
                        "@id": "viade:points"
                    },
                    "latitude": {
                        "@id": "schema:latitude",
                        "@type": "xs:double"
                    },
                    "longitude": {
                        "@id": "schema:longitude",
                        "@type": "xs:double"
                    },
                    "elevation": {
                        "@id": "schema:elevation",
                        "@type": "xsd:double"
                    },
                    "author": {
                        "@id": "schema:author",
                        "@type": "@id"
                    },
                    "rdf": "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
                    "rdfs": "http://www.w3.org/2000/01/rdf-schema#",
                    "schema": "http://schema.org/",
                    "viade": "http://arquisoft.github.io/viadeSpec/",
                    "xsd": "http://www.w3.org/2001/XMLSchema#"
                },
                "name": this.name,
                "author": this.author,
                "description": this.description,
                "comments": this.comments,
                "media": this.media,
                "points": routePointsJson
            }
        );
    }

}

/**
 * Method that receives a list of points and returns
 * a list of RouteElements
 * @param {Array} points List of points containing name, latitude, longitude, elevation. 
 */
function generateRouteElements(points) {
    let routeElements = [];
    points.forEach((p) => routeElements.push(new RouteElement(p.lat, p.lng)));
    return routeElements;
}

export default Route;