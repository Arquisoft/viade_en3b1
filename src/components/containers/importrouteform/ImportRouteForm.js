import React, { Component } from 'react';
import NavBar from '../../ui/main/NavBar';
import Footer from '../../ui/main/Footer';
import { withStyles, Typography, Paper, Grid, Button } from '@material-ui/core';
import Route from '../../../entities/Route';
import TrackPoint from '../../../entities/TrackPoint';
import ImportRouteCard from '../../ui/ImportRouteCard';

export class ImportRouteForm extends Component {

    constructor(props) {
        super(props);
        // console.log(this.props.location.routes);
        this.state = {
            // routes: this.props.location.routes,
            routes: [new Route("San Silvestre", "Esto es una ruta de prueba", [new TrackPoint(43, 5, 10), new TrackPoint(47, 8, 10)], null, null, null, null, null),
            new Route("Recorrido por Oviedo", "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.", [new TrackPoint(41, 5, 10), new TrackPoint(43, 8, 10)], null, null, null, null, null)]

        }
    }

    render() {
        const { classes } = this.props;
        const { routes } = this.state;
        return (
            <div>
                <NavBar />

                <Grid container className={classes.background}>
                    <Grid
                        item
                        xs={false}
                        sm={12}
                        md={12}
                        className={classes.image}>

                        <main className={classes.layout}>
                            <Paper className={classes.paper}>

                                <Typography component="h1" variant="h4" align="center">
                                    Please, review your routes
                                </Typography>

                                {routes.map((each) => (
                                    <ImportRouteCard route={each} />
                                ))}

                                <div className={classes.btnArea} >
                                    <Button className={classes.btn} color="primary" variant="contained">Accept</Button>
                                </div>
                            </Paper>
                        </main>

                    </Grid>

                    <Grid
                        item
                        xs={12}
                        sm={12}
                        md={12}
                    // style={{marginTop: '30rem'}}
                    >
                        <Footer />
                    </Grid>
                </Grid>
            </div>
        )
    }
}

const useStyles = (theme) => ({
    layout: {
        width: 'auto',
        marginLeft: theme.spacing(2),
        marginRight: theme.spacing(2),
        [theme.breakpoints.up(600 + theme.spacing(2) * 2)]: {
            width: 600,
            marginLeft: 'auto',
            marginRight: 'auto',
        },
    },
    paper: {
        marginTop: theme.spacing(3),
        marginBottom: theme.spacing(3),
        padding: theme.spacing(2),
        [theme.breakpoints.up(600 + theme.spacing(3) * 2)]: {
            marginTop: theme.spacing(6),
            marginBottom: theme.spacing(6),
            padding: theme.spacing(3),
        },
    },
    btn: {
        // marginRight: theme.spacing(50),
        marginTop: theme.spacing(5),
    },
    btnArea: {
        textAlign: 'center'
    },
    image: {
        backgroundImage: `url(https://source.unsplash.com/collection/9992041/1600x900)`,
        backgroundRepeat: 'no-repeat',
        backgroundColor:
            theme.palette.type === 'light' ? theme.palette.grey[50] : theme.palette.grey[900],
        backgroundSize: 'cover',
        backgroundPosition: 'center',
        minHeight: '90vh',
    },
});

export default withStyles(useStyles)(ImportRouteForm);
