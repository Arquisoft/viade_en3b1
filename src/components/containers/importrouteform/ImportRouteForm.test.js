import ImportRouteForm from './ImportRouteForm';
import { Router } from 'react-router';
import { getByTestId, fireEvent, render } from '@testing-library/react';
import RouteJSON from '../../../test-assets/RouteJSON.json';
import RouteGPX from '../../../test-assets/RouteGPX.gpx';
import Route from '../../../entities/Route';
import TrackPoint from '../../../entities/TrackPoint';

var file = new File(["foo"], "foo.txt", {
    type: "text/plain",
});

var route1 = new Route("Route1", "Description 1", [new TrackPoint(1,1), new TrackPoint(2,1)], null, null, null, "ID1");

test('Define ImportRouteForm', () => {
    expect(ImportRouteForm).toBeDefined();
});

test('Renders correctly', () => {
    const wrapper = render(<ImportRouteForm location={{ routes: [file] }} />);
    expect(wrapper).toMatchSnapshot();
});

test('Renders correctly: No files', () => {
    const wrapper = render(<ImportRouteForm location={{ routes: [] }} />);
    expect(wrapper).toMatchSnapshot();
});

test('Mounts correctly', () => {
    const wrapper = mount(<ImportRouteForm location={{ routes: [file] }} />);
    expect(wrapper).toBeTruthy();

    const wrapper2 = mount(<ImportRouteForm location={{ routes: [] }} />);
    expect(wrapper2).toBeTruthy();
});

test("Click", () => {
    const { container } = render(<ImportRouteForm location={{ routes: [file] }} />);

    const button = getByTestId(container, 'btn-test-accept');
    fireEvent.click(button);
});

test('Methods', () => {
    const wrapper = shallow(<ImportRouteForm location={{ routes: [file] }} />);
    const instance = wrapper.shallow().instance();

    instance.handleFiles();
    // instance.handleGPX(RouteGPX);
    // instance.handleJSON(RouteJSON);
});