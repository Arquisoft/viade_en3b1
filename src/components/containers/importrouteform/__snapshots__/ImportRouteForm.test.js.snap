// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="makeStyles-root-6"
        >
          <header
            class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-7 MuiToolbar-gutters"
            >
              <div
                class="MuiGrid-root makeStyles-link-9"
              >
                <div
                  class="MuiGrid-root MuiGrid-container"
                >
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
                    href="#/home"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item"
                      >
                        <img
                          alt="Viade Logo"
                          src="logo_2.svg"
                          style="height: 2.5rem; width: 2.5rem;"
                        />
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item"
                      >
                        <h4
                          class="MuiTypography-root MuiTypography-h4"
                        >
                          ViaDe
                        </h4>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-menuButton-8 MuiButton-containedPrimary"
                href="#/register"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  Sign Up
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-menuButton-8 MuiButton-colorInherit"
                data-testid="btn-login-navbar"
                href="#/login"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  Sign In
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
          </header>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root ImportRouteForm-image-5 MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
          >
            <main
              class="ImportRouteForm-layout-1"
            >
              <div
                class="MuiPaper-root ImportRouteForm-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
              >
                <h1
                  class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
                >
                  Please, review your routes
                </h1>
                <div
                  class="ImportRouteForm-btnArea-4"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained ImportRouteForm-btn-3 MuiButton-containedPrimary"
                    data-testid="btn-test-accept"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Accept
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </main>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
          >
            <div>
              <footer
                class="makeStyles-footer-226"
              >
                <div
                  class="MuiContainer-root MuiContainer-maxWidthSm"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6"
                  >
                    ViaDe En3B1
                  </h6>
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                  >
                    Made by Pablo Ca√±al
                  </h6>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                    href="https://github.com/Arquisoft/viade_en3b1/"
                    target="_blank"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                      />
                    </svg>
                  </a>
                </div>
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="makeStyles-root-6"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-7 MuiToolbar-gutters"
          >
            <div
              class="MuiGrid-root makeStyles-link-9"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
                  href="#/home"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <img
                        alt="Viade Logo"
                        src="logo_2.svg"
                        style="height: 2.5rem; width: 2.5rem;"
                      />
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <h4
                        class="MuiTypography-root MuiTypography-h4"
                      >
                        ViaDe
                      </h4>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-menuButton-8 MuiButton-containedPrimary"
              href="#/register"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Sign Up
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-menuButton-8 MuiButton-colorInherit"
              data-testid="btn-login-navbar"
              href="#/login"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Sign In
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </header>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root ImportRouteForm-image-5 MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
        >
          <main
            class="ImportRouteForm-layout-1"
          >
            <div
              class="MuiPaper-root ImportRouteForm-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h1
                class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
              >
                Please, review your routes
              </h1>
              <div
                class="ImportRouteForm-btnArea-4"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained ImportRouteForm-btn-3 MuiButton-containedPrimary"
                  data-testid="btn-test-accept"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Accept
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </main>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
        >
          <div>
            <footer
              class="makeStyles-footer-226"
            >
              <div
                class="MuiContainer-root MuiContainer-maxWidthSm"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6"
                >
                  ViaDe En3B1
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                >
                  Made by Pablo Ca√±al
                </h6>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="https://github.com/Arquisoft/viade_en3b1/"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                    />
                  </svg>
                </a>
              </div>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders correctly: No files 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="makeStyles-root-259"
        >
          <header
            class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-260 MuiToolbar-gutters"
            >
              <div
                class="MuiGrid-root makeStyles-link-262"
              >
                <div
                  class="MuiGrid-root MuiGrid-container"
                >
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
                    href="#/home"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item"
                      >
                        <img
                          alt="Viade Logo"
                          src="logo_2.svg"
                          style="height: 2.5rem; width: 2.5rem;"
                        />
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item"
                      >
                        <h4
                          class="MuiTypography-root MuiTypography-h4"
                        >
                          ViaDe
                        </h4>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-menuButton-261 MuiButton-containedPrimary"
                href="#/register"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  Sign Up
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-menuButton-261 MuiButton-colorInherit"
                data-testid="btn-login-navbar"
                href="#/login"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  Sign In
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
          </header>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root ImportRouteForm-image-258 MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
          >
            <main
              class="ImportRouteForm-layout-254"
            >
              <div
                class="MuiPaper-root ImportRouteForm-paper-255 MuiPaper-elevation1 MuiPaper-rounded"
              >
                <h1
                  class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
                >
                  Please, review your routes
                </h1>
                <div
                  class="ImportRouteForm-btnArea-257"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained ImportRouteForm-btn-256 MuiButton-containedPrimary"
                    data-testid="btn-test-accept"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Accept
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </main>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
          >
            <div>
              <footer
                class="makeStyles-footer-479"
              >
                <div
                  class="MuiContainer-root MuiContainer-maxWidthSm"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6"
                  >
                    ViaDe En3B1
                  </h6>
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                  >
                    Made by Pablo Ca√±al
                  </h6>
                  <a
                    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                    href="https://github.com/Arquisoft/viade_en3b1/"
                    target="_blank"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                      />
                    </svg>
                  </a>
                </div>
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="makeStyles-root-259"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-260 MuiToolbar-gutters"
          >
            <div
              class="MuiGrid-root makeStyles-link-262"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
                  href="#/home"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <img
                        alt="Viade Logo"
                        src="logo_2.svg"
                        style="height: 2.5rem; width: 2.5rem;"
                      />
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <h4
                        class="MuiTypography-root MuiTypography-h4"
                      >
                        ViaDe
                      </h4>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-menuButton-261 MuiButton-containedPrimary"
              href="#/register"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Sign Up
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-menuButton-261 MuiButton-colorInherit"
              data-testid="btn-login-navbar"
              href="#/login"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Sign In
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </header>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root ImportRouteForm-image-258 MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
        >
          <main
            class="ImportRouteForm-layout-254"
          >
            <div
              class="MuiPaper-root ImportRouteForm-paper-255 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h1
                class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
              >
                Please, review your routes
              </h1>
              <div
                class="ImportRouteForm-btnArea-257"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained ImportRouteForm-btn-256 MuiButton-containedPrimary"
                  data-testid="btn-test-accept"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Accept
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </main>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
        >
          <div>
            <footer
              class="makeStyles-footer-479"
            >
              <div
                class="MuiContainer-root MuiContainer-maxWidthSm"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6"
                >
                  ViaDe En3B1
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                >
                  Made by Pablo Ca√±al
                </h6>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                  href="https://github.com/Arquisoft/viade_en3b1/"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                    />
                  </svg>
                </a>
              </div>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
